homeassistant/components/group/climate.py

```diff
@@ -0,0 +1,396 @@
+
+    def __init__(
+        self,
+        unique_id: str | None,
+        name: str,
+        entity_ids: list[str],
+        temperature_unit: str,
+    ) -> None:
+        """Initialize a climate group."""
+        self._name = name
+        self._entity_ids = entity_ids
+
+        self._features: dict[ClimateEntityFeature, set[str]] = {
+            (
+                ClimateEntityFeature.TARGET_TEMPERATURE
+                | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            ): set(),
+            ClimateEntityFeature.TARGET_HUMIDITY: set(),
+            ClimateEntityFeature.FAN_MODE: set(),
+            ClimateEntityFeature.PRESET_MODE: set(),
+            ClimateEntityFeature.SWING_MODE: set(),
+            ClimateEntityFeature.AUX_HEAT: set(),
+            ClimateEntityFeature.TURN_ON: set(),
+            ClimateEntityFeature.TURN_OFF: set(),
+        }
+
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+        self._attr_extra_state_attributes = {ATTR_ENTITY_ID: entity_ids}
+
+        self._attr_temperature_unit = temperature_unit
+
+        # Set some defaults (will be overwritten on update)
+        self._attr_supported_features: ClimateEntityFeature = ClimateEntityFeature(0)
+        self._attr_hvac_modes: list[HVACMode] = [HVACMode.OFF]
+        self._attr_hvac_mode: HVACMode | None = None
+        self._attr_hvac_action: HVACAction | None = None
+
+        self._attr_swing_modes: list[str] | None = None
+        self._attr_swing_mode: str | None = None
+
+        self._attr_fan_modes: list[str] | None = None
+        self._attr_fan_mode: str | None = None
+
+        self._attr_preset_modes: list[str] | None = None
+        self._attr_preset_mode: str | None = None
+
+    @callback
+    def async_update_supported_features(
+        self,
+        entity_id: str,
+        new_state: State | None,
+    ) -> None:
+        """Update dictionaries with supported features."""
+        if not new_state:
+            for climate in self._features.values():
+                climate.discard(entity_id)
+            return
+
+        new_features = new_state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+        for feature_flag, entity_set in self._features.items():
+            if new_features & feature_flag:
+                entity_set.add(entity_id)
+            else:
+                entity_set.discard(entity_id)
+
+    @callback
+    def async_update_group_state(self) -> None:
+        """Query all members and determine the climate group state."""
+        self._attr_assumed_state = False
+
+        states = [
+            state
+            for entity_id in self._entity_ids
+            if (state := self.hass.states.get(entity_id)) is not None
+        ]
+        self._attr_assumed_state |= not states_equal(states)
+
+        # Set group as unavailable if all members are unavailable or missing
+        self._attr_available = any(state.state != STATE_UNAVAILABLE for state in states)
```

Comment:
If it's unavailable we can probably bail out early?

Reply:
Checking at the rest of group implementations there is no early return in case the entity is marked as unavailable. Also conceptually, availability and working attributes of the group entities are independent attributes (if unavailable the entity should not be marked as not having any other attribute).